src/dmd/mtype.d

```diff
@@ -2056,8 +2056,8 @@ extern (C++) abstract class Type : ASTNode
         if (s.isAliasDeclaration())
             s = s.toAlias();
 
-        if (s.isTupleDeclaration())
-            return null;
+        if (TupleDeclaration td = s.isTupleDeclaration())
+            return td.tupletype;
```

Comment:
You should use td.getType() but still this won't work with tuples that have values (not only types).

Reply:
And because the real problem in your test case has nothing to do with `alias this`, so this code is wrong.